### PR TITLE
tf script for ranger2 with asg

### PR DESCRIPTION
**Pull request by pritishsahoo**
tf script for ranger2 with asg

### CODE REVIEW

File: ranger2/aws/ranger_asg/providers.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#Configure AWS Provider
+provider "aws" {
+  access_key = var.access_key
+  secret_key = var.secret_key
+  region     = var.region
+  token 	 = var.aws_session_token
```
Comment (vihag) [inline ranger2/aws/ranger_asg/providers.tf]: out of curiosity - i understand that the session tokens are required for devexternalqubole. if the customer using it does not have the same security settings, will this cause any authorization errors
?

File: ranger2/aws/ranger_asg/ranger_bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+source /usr/lib/qubole/bootstrap-functions/hive/ranger-client.sh
+
+#Usage for any Hive Cluster: install_ranger -h "RangerDNS/LB" -p Port -S "SolrDNS/LB" -P port
+install_ranger -h xxxx-ranger-alb-xxxxxx.us-east-1.elb.amazonaws.com -p 80 -S xxx-solr-alb-xxxx.us-east-1.elb.amazonaws.com -P 80
```
Comment (vihag) [inline ranger2/aws/ranger_asg/ranger_bootstrap.sh]: should we be hardcoding us-east-1 here? I think the region should be configurable(not even take from the session credentials)

File: ranger2/aws/ranger_asg/ranger_policy.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+#Loadbalancer Port
+#Usage: python ranger_policy.py LB-IP pritish-qubole hive_policy 80
+
+host = sys.argv[1]
+defloc_loc =  sys.argv[2] 
+service_name = sys.argv[3]
+port = sys.argv[4]
+
+#host = "ps-cvs-ranger-asg-alb-1733501465.us-east-1.elb.amazonaws.com"
+#defloc_loc =  "pritish-qubole"
```
Comment (vihag) [inline ranger2/aws/ranger_asg/ranger_policy.py]: we can have descriptive comments on where to get this information from.

File: ranger2/aws/ranger_asg/ranger_policy.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import requests
+import json, sys
```
Comment (vihag) [inline ranger2/aws/ranger_asg/ranger_policy.py:2]: there are quite a lot of tpl and py files here. Have we seen what happens when update the terraform configuration? say you perform terraform apply - everything runs fine and deploys. next you want to change something, say some subnet somewhere, will these policy py files run again? if so, will it cause duplicate policies or inconsistencies?

File: ranger2/aws/ranger_asg/ranger_policy.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    res_data = json.loads(res.text)
+    if(res.status_code == 200):
+        print('Created service successfully: ' + service_name )
+    else:
+        print('Failed creating Ranger Repository: ' + str(res_data))
+        sys.exit(1)
+
+# Create qbol_user
+def create_qbl_usr(qbl_user):
+    qbol_user_json = {"name": qbl_user,"firstName": qbl_user,"password":"Ranger123","userRoleList":["ROLE_ADMIN_AUDITOR"]}
```
Comment (vihag) [inline ranger2/aws/ranger_asg/ranger_policy.py]: we should probably convert this file to a tpl file and make these configurable. hardcoded passwords might not sit well with folks.

File: ranger2/aws/ranger_asg/solr_ami.tpl [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+cd $SOLR_INSTALL
+
+wget https://dist.apache.org/repos/dist/release/ranger/${RANGER_VER}/apache-ranger-${RANGER_VER}.tar.gz
+tar -zxf apache-ranger-${RANGER_VER}.tar.gz
+rm apache-ranger-${RANGER_VER}.tar.gz
+cd apache-ranger-${RANGER_VER}/security-admin/contrib/solr_for_audit_setup
+pwd
+
+sed -i 's/SOLR_INSTALL=false/SOLR_INSTALL=true/g' install.properties
+sed -i 's/SOLR_MAX_MEM=2g/SOLR_MAX_MEM=512m/g' install.properties
```
Comment (vihag) [inline ranger2/aws/ranger_asg/solr_ami.tpl]: should mem settings be configurable?

File: ranger2/aws/ranger_asg/terraform.tfvars [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+aws_session_token = "XXXX"
+
+region = "us-east-1"
+
+vpc_id = "vpc-XXXX"
+public_subnets = ["subnet-XXXX", "subnet-XXXX"]
+private_subnets = ["subnet-XXXX", "subnet-XXXX"]
+#your local ip
+local_ips = ["99.0.0.0/32"]
```
Comment (vihag) [inline ranger2/aws/ranger_asg/terraform.tfvars:12]: whose local IP is this ? the person who is setting this up ?

File: ranger2/aws/ranger_asg/main.tf [hunk 8]
```diff
@@ -0,0 +1,579 @@
...
+    Name = var.ranger_solr_ami_name
+  }
+  user_data = data.template_file.solr_base_ami_tmpl.rendered
+
+  provisioner "remote-exec" {
+    connection {
+      type        = "ssh"
+      user        = "ec2-user"
+      host        = aws_instance.tf_solr_base_inst.public_dns
+      private_key = file("${path.module}/pskey.pem")
```
Comment (vihag) [inline ranger2/aws/ranger_asg/main.tf]: pskey.pem -> is this pritishsahookey.pem :) maybe we can make this more explicit that this needs to be configured

File: ranger2/aws/ranger_asg/main.tf [hunk 9]
```diff
@@ -0,0 +1,579 @@
...
+    }
+
+    inline = [
+      "sleep 120;while [ ! -f /tmp/setup-done ]; do echo 'Waiting for solr setup script to complete'; sleep 20; done;"
```
Comment (vihag) [inline ranger2/aws/ranger_asg/main.tf]: what happens in case of failure, will this be in a forever loop?

File: ranger2/aws/ranger_asg/main.tf [hunk 10]
```diff
@@ -0,0 +1,579 @@
...
+  tags = { 
+    Name = var.ranger_admin_ami_name
+  }
+
+  provisioner "remote-exec" {
+    connection {
+      type        = "ssh"
+      user        = "ec2-user"
+      host        = aws_instance.tf_ranger_base_inst.public_dns
+      private_key = file("${path.module}/pskey.pem")
```
Comment (vihag) [inline ranger2/aws/ranger_asg/main.tf]: same as the other pem file name

File: ranger2/aws/ranger_asg/main.tf [hunk 11]
```diff
@@ -0,0 +1,579 @@
...
+      key = "Name"
+      value = var.ranger_admin_name
+      propagate_at_launch = true
+    }
+}
+
+resource "null_resource" "create-default-ranger-policy" {
+    depends_on = [aws_autoscaling_group.tf_ranger_asg]
+    provisioner "local-exec" {
+      command = "sleep 240;python ranger_policy.py ${aws_lb.ranger_alb.dns_name} ${var.def_loc} ${var.service_name} ${var.ranger_alb_port}"
```
Comment (vihag) [inline ranger2/aws/ranger_asg/main.tf]: how will we catch failures in the py file execution?

File: ranger2/aws/ranger_asg/providers.tf (hunk 1 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/providers.tf]: Fixed added condition

File: ranger2/aws/ranger_asg/ranger_bootstrap.sh (hunk 2 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/ranger_bootstrap.sh]: Fixed added comments

File: ranger2/aws/ranger_asg/ranger_policy.py (hunk 3 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/ranger_policy.py]: Fixed and added comments.

File: ranger2/aws/ranger_asg/ranger_policy.py (hunk 4 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/ranger_policy.py:2]: No py file will not rerun. Script checks for exist condition before creating.

File: ranger2/aws/ranger_asg/ranger_policy.py (hunk 5 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/ranger_policy.py]: Fixed. Added config parameter

File: ranger2/aws/ranger_asg/terraform.tfvars (hunk 7 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/terraform.tfvars:12]: This IPs will be connecting AWS resource from like LBs or ssh into instance.

File: ranger2/aws/ranger_asg/main.tf (hunk 10 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/main.tf]: Fixed and added config.

File: ranger2/aws/ranger_asg/main.tf (hunk 11 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/main.tf]: Exceptions will be displayed on the console

File: ranger2/aws/ranger_asg/main.tf (hunk 9 above)
Comment (pritishsahoo) [reply ranger2/aws/ranger_asg/main.tf]: Fixed added logic to exit

**Review by pritishsahoo — COMMENTED** [review]
Comments Fixed

**Review by vihag — APPROVED** [review]

**Pull request merged**